content/docs/guides/components/istio.md

```diff
@@ -0,0 +1,162 @@
+
+### Metrics
+The istio sidecar reports data to [Mixer](https://istio.io/docs/concepts/policy-and-control/mixer.html).
+We can view the istio dashboard by [installing Grafana](https://istio.io/docs/tasks/telemetry/using-istio-dashboard.html#viewing-the-istio-dashboard).
+Execute the command:
+```
+kubectl -n istio-system port-forward $(kubectl -n istio-system get pod -l app=grafana -o jsonpath='{.items[0].metadata.name}') 3000:3000 &
+```
+Visit http://localhost:3000/dashboard/db/istio-dashboard in your web browser.
+Send some requests to the TF serving service, then there should be some data (QPS, success rate, latency)
+like
+![istio dashboard](../istio-dashboard.png)
+
+
+#### Define and view metrics
+See istio [doc](https://istio.io/docs/tasks/telemetry/metrics-logs.html).
+
+#### Expose Grafana dashboard behind ingress/IAP
+To expose the grafana dashboard as, e.g. `YOUR_HOST/grafana`, follow these steps.
+
+  1. Add ambassador annotation for routing. However, since ambassador only scans the service within
+  its [namespace](https://www.getambassador.io/reference/advanced),
+  we can add the annotation for grafana service in ambassador service. So do
+  `kubectl edit svc -n kubeflow ambassador`, and add annotation
+
+  ```
+  getambassador.io/config: |
+    ---
+    apiVersion: ambassador/v0
+    kind:  Mapping
+    name:  grafana_dashboard_mapping
+    prefix: /grafana/
+    service: grafana.istio-system:3000
+  ```
+
+  2. Grafana needs to be [configured](http://docs.grafana.org/installation/behind_proxy/#examples-with-sub-path-ex-http-foo-bar-com-grafana)
+  to work properly behind a reverse proxy. We can override the default config using
+  [environment variable](http://docs.grafana.org/installation/configuration/#using-environment-variables).
+  So do `kubectl edit deploy -n istio-system grafana`, and add env vars
+  ```
+  - name: GF_SERVER_DOMAIN
+    value: YOUR_HOST
+  - name: GF_SERVER_ROOT_URL
+    value: '%(protocol)s://%(domain)s:/grafana'
+  ```
+
+### Rolling out new model
+
+A typical scenario is that we first deploy a model A. Then we develop another model B, and we want to deploy it
+and gradually move traffic from A to B. This can be achieved using Istio's traffic routing.
+
+  1. Deploy the first model as [usual](README.md), with name X. We can optionally set the version param, which
+  default to `v1`. After doing `ks apply`, we will have a service X, and a deployment `X-v1`.
+  In addition, a default routing rule is created and routes all requests to `v1`.
+  2. When we want to rollout the new model, use the same name X and set a different version, e.g. `v2`.
+  ```
+  ks param set --env=$ENV $MODEL_COMPONENT version v2
+  ks param set --env=$ENV $MODEL_COMPONENT firstVersion false
+  ks apply $ENV -c $MODEL_COMPONENT
+  ```
+  This deploys the new deployment `X-v2`, but the traffic will still go to `v1`
+  3. Create the new routing rule. For example, the following sends 5% traffic to `v2`.
+  ```
```

Comment:
This pre section is not rendering properly. Similarly the one above lines 138-142. You might want to try playing around by leaving an empty line before/after or both.

Reply:
Thanks @abhi-g. This Hugo creates a lot of markdown incompatibility :-) updated PR.